src/Dfc.CourseDirectory.WebV2/Features/TLevels/ViewAndEditTLevel/EditTLevel.cs

```diff
@@ -0,0 +1,199 @@
+                state.Website = request.Website;
+                state.WhoFor = request.WhoFor;
+                state.EntryRequirements = request.EntryRequirements;
+                state.WhatYoullLearn = request.WhatYoullLearn;
+                state.HowYoullBeAssessed = request.HowYoullBeAssessed;
+                state.HowYoullLearn = request.HowYoullLearn;
+                state.WhatYouCanDoNext = request.WhatYouCanDoNext;
+                state.IsValid = true;
+            });
+
+            return new Success();
+        }
+
+        private async Task CheckNoExistingTLevelForStartDate(ValidationResult validationResult, Command command)
+        {
+            if (!command.StartDate.HasValue)
+            {
+                return;
+            }
+
+            var existingTLevels = await _sqlQueryDispatcher.ExecuteQuery(
+                new GetTLevelsForProvider() { ProviderId = _journeyInstance.State.ProviderId });
+
+            if (existingTLevels.Any(tl =>
+                tl.TLevelDefinition.TLevelDefinitionId == _journeyInstance.State.TLevelDefinitionId &&
+                tl.TLevelId != _journeyInstance.State.TLevelId &&
+                tl.StartDate == command.StartDate?.ToDateTime()))
+            {
+                validationResult.Errors.Add(
+                    new ValidationFailure(nameof(Command.StartDate), "Start date already exists"));
+            }
+        }
+
+        private ViewModel CreateViewModel(IReadOnlyCollection<Venue> providerVenues) => new ViewModel()
+        {
+            ProviderVenues = providerVenues
+                .Select(v => new ViewModelProviderVenuesItem()
+                {
+                    VenueId = v.VenueId,
+                    VenueName = v.VenueName
+                })
+                .ToList(),
+            LocationVenueIds = new HashSet<Guid>(_journeyInstance.State.LocationVenueIds),
+            StartDate = (Date?)_journeyInstance.State.StartDate,
+            Website = _journeyInstance.State.Website,
+            YourReference = _journeyInstance.State.YourReference,
+            EntryRequirements = _journeyInstance.State.EntryRequirements,
+            HowYoullBeAssessed = _journeyInstance.State.HowYoullBeAssessed,
+            HowYoullLearn = _journeyInstance.State.HowYoullLearn,
+            TLevelId = _journeyInstance.State.TLevelId,
+            TLevelName = _journeyInstance.State.TLevelName,
+            WhatYouCanDoNext = _journeyInstance.State.WhatYouCanDoNext,
+            WhatYoullLearn = _journeyInstance.State.WhatYoullLearn,
+            WhoFor = _journeyInstance.State.WhoFor
+        };
+
+        private Task<IReadOnlyCollection<Venue>> GetVenuesForProvider() =>
+            _sqlQueryDispatcher.ExecuteQuery(
+                new GetVenuesForProvider()
+                {
+                    ProviderId = _journeyInstance.State.ProviderId
+                });
+    }
+
+    public class CommandValidator : ValidatorBase<Command>
```

Comment:
Does this need to be `public`?

Reply:
Probably not, but most of our validators are `public`

Reply:
I don't think that's a very good argument. If it doesn't need to be publicly accessible, it shouldn't be.